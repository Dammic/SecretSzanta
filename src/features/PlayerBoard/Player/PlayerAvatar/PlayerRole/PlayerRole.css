@import 'variables/variables.css';

.ribbon {
    position: absolute;
    right: -5px;
    top: -5px;
    z-index: 1;
    overflow: hidden;
    width: 75px;
    height: 75px;
    text-align: right;
    pointer-events: none;
}

.roleName {
    font-size: 10px;
    font-weight: bold;
    line-height: 20px;
    color: #FFF;
    text-transform: uppercase;
    text-align: center;
    transform: rotate(45deg);
    width: 100px;
    display: block;
    background: linear-gradient(#9BC90D 0%, #79A70A 100%);
    box-shadow: 0 3px 10px -5px rgba(0, 0, 0, 1);
    position: absolute;
    top: 19px;
    right: -21px;
}

.roleName::after, .roleName::before {
    content: "";
    position: absolute;
    top: 100%;
    z-index: -1;
    border-bottom: 3px solid transparent;
    border-top: 3px solid #79A70A;
}

.roleName::before {
    left: 0px;
    border-left: 3px solid #79A70A;
    border-right: 3px solid transparent;
}

.roleName::after {
    right: 0px;
    border-left: 3px solid transparent;
    border-right: 3px solid #79A70A;
}

.chancellor {
    background: linear-gradient(#CB5B5B 0%, #843636 100%);

    &::after, &::before {
        border-top: 3px solid #843636;
    }

    &::after {
        border-right: 3px solid #843636;
    }

    &::before {
        border-left: 3px solid #843636;
    }
}

.previous {
    background: linear-gradient(#8CCACF 0%, #7C9296 100%);

    &::after, &::before {
        border-top: 3px solid #7C9296;
    }

    &::after {
        border-right: 3px solid #7C9296;
    }

    &::before {
        border-left: 3px solid #7C9296;
    }
}
