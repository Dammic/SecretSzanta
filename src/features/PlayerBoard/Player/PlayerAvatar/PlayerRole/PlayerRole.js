import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { PlayerRole as PlayerRoleDict } from '../../../../../../Dictionary'

import styles from './PlayerRole.css'

const roleMapping = {
    [PlayerRoleDict.ROLE_CHANCELLOR]: 'Chancellor',
    [PlayerRoleDict.ROLE_PRESIDENT]: 'President',
    [PlayerRoleDict.ROLE_PREVIOUS_CHANCELLOR]: 'P. Chancellor',
    [PlayerRoleDict.ROLE_PREVIOUS_PRESIDENT]: 'P. President',
}

const PlayerRole = ({ role }) => {
    if (!role || !roleMapping[role]) {
        return null
    }

    const roleName = roleMapping[role]
    const ribbonNameClassNames = classNames(styles.roleName, {
        [styles.chancellor]: role === PlayerRoleDict.ROLE_CHANCELLOR,
    })

    return (
        <div className={styles.ribbon}>
            <span className={ribbonNameClassNames}>
                {roleName}
            </span>
        </div>
    )
}

PlayerRole.displayName = 'PlayerRole'
PlayerRole.propTypes = {
    role: PropTypes.string,
}

export { PlayerRole }
export default PlayerRole
