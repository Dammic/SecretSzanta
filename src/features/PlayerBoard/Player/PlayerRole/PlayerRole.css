@import 'variables/variables.css';

.role {
    width: var(--player-image-width);
    max-width: 130px;
}
