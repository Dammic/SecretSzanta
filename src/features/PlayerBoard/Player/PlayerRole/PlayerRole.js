import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { PlayerRole as PlayerRoleDict } from '../../../../../Dictionary'

import roleChancellorPicture from '../../../../static/Chancellor.png'
import rolePresidentPicture from '../../../../static/President.png'
import rolePreviousChancellorPicture from '../../../../static/PreviousChancelor.png'
import rolePreviousPresidentPicture from '../../../../static/PreviousPresident.png'

import styles from './PlayerRole.css'

const roleImagesMapping = {
    [PlayerRoleDict.ROLE_CHANCELLOR]: roleChancellorPicture,
    [PlayerRoleDict.ROLE_PRESIDENT]: rolePresidentPicture,
    [PlayerRoleDict.ROLE_PREVIOUS_CHANCELLOR]: rolePreviousChancellorPicture,
    [PlayerRoleDict.ROLE_PREVIOUS_PRESIDENT]: rolePreviousPresidentPicture,
}

const PlayerRole = ({ role, className }) => {
    if (!role || !roleImagesMapping[role]) {
        return null
    }

    const rolePicture = roleImagesMapping[role]
    return (
        <img className={classNames(styles.role, className)} src={rolePicture} alt={role} />
    )
}

PlayerRole.displayName = 'PlayerRole'
PlayerRole.propTypes = {
    role: PropTypes.string,
    className: PropTypes.string,
}
export default PlayerRole
