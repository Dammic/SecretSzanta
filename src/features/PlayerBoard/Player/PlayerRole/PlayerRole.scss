.role {
    width: $player-image-width;
    max-width: 130px;
}
