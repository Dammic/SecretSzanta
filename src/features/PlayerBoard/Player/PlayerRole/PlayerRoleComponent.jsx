import React from 'react'
import PropTypes from 'prop-types'
import { PlayerRole } from '../../../../../Dictionary'

import styles from './PlayerRole.css'

const roleChancellorPicture = require('../../../../static/Chancellor.png')
const rolePresidentPicture = require('../../../../static/President.png')
const rolePreviousChancellorPicture = require('../../../../static/PreviousChancelor.png')
const rolePreviousPresidentPicture = require('../../../../static/PreviousPresident.png')

const PlayerRoleComponent = ({
    role,
}) => {
    let rolePicture
    switch (role) {
        case PlayerRole.ROLE_CHANCELLOR:
            rolePicture = roleChancellorPicture
            break
        case PlayerRole.ROLE_PRESIDENT:
            rolePicture = rolePresidentPicture
            break
        case PlayerRole.ROLE_PREVIOUS_CHANCELLOR:
            rolePicture = rolePreviousChancellorPicture
            break
        case PlayerRole.ROLE_PREVIOUS_PRESIDENT:
            rolePicture = rolePreviousPresidentPicture
            break
        default:
            rolePicture = ''
    }
    return (role ? <img className={styles.role} src={rolePicture} alt="" /> : null)
}

PlayerRoleComponent.propTypes = {
    role: PropTypes.string,
}
export default PlayerRoleComponent
