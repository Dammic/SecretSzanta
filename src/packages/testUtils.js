import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
