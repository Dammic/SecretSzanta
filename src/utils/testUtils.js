import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
