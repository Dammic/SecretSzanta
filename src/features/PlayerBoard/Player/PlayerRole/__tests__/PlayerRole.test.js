import React from 'react'
import { shallow } from 'enzyme'
import { forEach } from 'lodash'
import { expectMatchingSnapshot } from 'packages/testUtils'
import { PlayerRole as PlayerRoleDict } from '../../../../../../Dictionary'
import PlayerRole from '../PlayerRole'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        role: null,
        className: 'mockClassName',
        ...propsOverrides,
    }
    const component = renderMethod(<PlayerRole {...props} />)
    return { props, component }
}

describe('<PlayerRole />', () => {
    it('returns null if role is not defined', () => {
        const { props } = setupProps({ role: null })
        expectMatchingSnapshot(<PlayerRole {...props} />)
    })

    it('returns null if role is incorrect', () => {
        const { props } = setupProps({ role: 'someincorrectrole' })
        expectMatchingSnapshot(<PlayerRole {...props} />)
    })

    it('returns proper images if role is one of correct roles', () => {
        const correctRoles = [
            PlayerRoleDict.ROLE_CHANCELLOR,
            PlayerRoleDict.ROLE_PRESIDENT,
            PlayerRoleDict.ROLE_PREVIOUS_CHANCELLOR,
            PlayerRoleDict.ROLE_PREVIOUS_PRESIDENT,
        ]
        forEach(correctRoles, (role) => {
            const { props } = setupProps({ role })
            expectMatchingSnapshot(<PlayerRole {...props} />)
        })
    })
})
