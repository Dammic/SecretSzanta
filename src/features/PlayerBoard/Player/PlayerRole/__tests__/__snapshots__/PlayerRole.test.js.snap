// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerRole /> returns null if role is incorrect 1`] = `null`;

exports[`<PlayerRole /> returns null if role is not defined 1`] = `null`;

exports[`<PlayerRole /> returns proper images if role is one of correct roles 1`] = `
<img
  alt="ROLE_CHANCELLOR"
  className="role mockClassName"
  src="test-file-stub"
/>
`;

exports[`<PlayerRole /> returns proper images if role is one of correct roles 2`] = `
<img
  alt="ROLE_PRESIDENT"
  className="role mockClassName"
  src="test-file-stub"
/>
`;

exports[`<PlayerRole /> returns proper images if role is one of correct roles 3`] = `
<img
  alt="ROLE_PREVIOUS_CHANCELLOR"
  className="role mockClassName"
  src="test-file-stub"
/>
`;

exports[`<PlayerRole /> returns proper images if role is one of correct roles 4`] = `
<img
  alt="ROLE_PREVIOUS_PRESIDENT"
  className="role mockClassName"
  src="test-file-stub"
/>
`;
